Add Lushlilac / Sea Lumies highlighter opacity option

DaysSky opened the pull request.

DaysSky committed:
Added Tree Break Progress HUD

DaysSky committed:
Fixed Translations for Tree Break HUD

DaysSky committed:
Add Lushlilac / Sea Lumies Highlighter Opacity

DaysSky committed:
fix merge conflict

DaysSky committed:
fixed formatting isues

DaysSky committed:
Fixed Tree Break Progress HUD Formatting

DaysSky committed:
actually make opacity option work

DaysSky committed:
Merge branch 'SkyblockerMod:master' into opacity

DaysSky committed:
Merge branch 'master' into opacity

contionability committed:
improve ability pattern (#1440)

PeytonBrown committed:
Fix ColorSelectionWidget showing on chain armor. (#1438)

AzureAaron committed:
Add up/down buttons to move waypoint indices

PeytonBrown commented on src/main/java/de/hysky/skyblocker/skyblock/galatea/TreeBreakProgressHud.java:
```diff
@@ -0,0 +1,121 @@
+		if (client.player == null || world == null)
+			return;
+
+		Stream<Entity> entityStream = StreamSupport.stream(world.getEntities().spliterator(), false);
```
Iterating over all world entities every frame can be expensive
can do `world.getEntitiesByClass(ArmorStandEntity.class, box, predicate)` to get a pre filtered list.

creating a cached list to store an entity list every 10-20 ticks might help improve performance.

PeytonBrown commented on src/main/java/de/hysky/skyblocker/skyblock/galatea/TreeBreakProgressHud.java:
```diff
@@ -0,0 +1,121 @@
+
+	@Override
+	public boolean shouldRender(Location location) {
+		return super.shouldRender(location) && true;
```
`&& true` does nothing

viciscat committed:
Item list things and forge!!!!!! (#948)

* item list shenanigans

* emi, finish forge and recipe book

* remove emi runtime

* Refactor recipe

* Render forge recipe duration

* Multiple things
Remove unused accessor
Remove REI runtime
Make EMI and recipe book call render method of a recipe

* work with 1.21.3

* REI priority + config

* requested changes and uh stuff
ye

* update REI

* fix stuff maybe

* fix wrapRenderer block clicks; lava bucket -> fred

---------

Co-authored-by: Kevinthegreat <92656833+kevinthegreat1@users.noreply.github.com>
Co-authored-by: Yasin <19829407+LifeIsAParadox@users.noreply.github.com>

kevinthegreat1 committed:
Add Spotless and Checkstyle (#1189)

* Add spotless

* Configure spotless

* Add checkstyle

* Allow some stuff

* Turn off tab enforcement

* Get rid of some more rules

* Remove indent rules

* Don't sort lang files

* Remove wrapping rules

* Apply spotless

* Apply spotless again

* Fix checkstyle violations

* Update checkstyle DTD

* Add Javadoc and Mixin checks

Alex33856 committed:
Fix Farming Hud Skill EXP not working when Health/Mana bars disabled (#1449)

Alex33856 committed:
Show Input Calculator on Bazaar flip input sign (#1457)

* Show Input Calculator on Bazaar flip input sign

* Use constants for strings

viciscat committed:
resourcepack compatibility (#1407)

* furfsky compatibility v1

* different title if quick nav is enabled in inventory screen

* oops

* change how slot is rendered

* fix slot and check for firmament first

* Update src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/SkyblockCraftingRecipeResults.java

Co-authored-by: Yasin <19829407+LifeIsAParadox@users.noreply.github.com>

* Update SkyblockCraftingRecipeResults.java

* Update SkyblockCraftingRecipeResults.java

* Update SkyblockCraftingRecipeResults.java

* edit id

* make it more general

---------

Co-authored-by: Yasin <19829407+LifeIsAParadox@users.noreply.github.com>

LifeIsAParadox committed:
Update SkyblockRecipeResults.java

DaysSky committed:
fixed tree break hud implementation

DaysSky committed:
fixed minor typo and tree break hud packet.

Right now the opacity config only works on game launch and I don't know how to fix it

DaysSky committed:
Merge remote-tracking branch 'upstream/master' into opacity

DaysSky committed:
Merge remote-tracking branch 'upstream/master' into opacity

DaysSky committed:
fix build error

DaysSky committed:
Delete .java-version

DaysSky committed:
Merge branch 'master' into opacity

kevinthegreat1 commented on src/main/java/de/hysky/skyblocker/skyblock/galatea/AbstractBlockHighlighter.java:
```diff
@@ -28,25 +28,29 @@
 	 */
-	protected AbstractBlockHighlighter(Block target, DyeColor colour) {
-		this(state -> state.isOf(target), colour);
+	protected AbstractBlockHighlighter(Block target, DyeColor colour, Float alpha) {
```
```suggestion
	protected AbstractBlockHighlighter(Block target, DyeColor colour, float alpha) {
```

kevinthegreat1 commented on src/main/java/de/hysky/skyblocker/skyblock/galatea/AbstractBlockHighlighter.java:
```diff
@@ -28,25 +28,29 @@
+	 * @param alpha Alpha for highlighting
 	 */
-	protected AbstractBlockHighlighter(Predicate<BlockState> statePredicate, DyeColor colour) {
+	protected AbstractBlockHighlighter(Predicate<BlockState> statePredicate, DyeColor colour, Float alpha) {
```
```suggestion
	protected AbstractBlockHighlighter(Predicate<BlockState> statePredicate, DyeColor colour, float alpha) {
```

kevinthegreat1 reviewed (commented):
Depends on #1403

DaysSky committed:
review changs

DaysSky committed:
Merge branch 'master' into opacity

DaysSky committed:
checkstyle fix

Alex33856 reviewed (changes requested):
Can you make it update the alpha value when the Config value gets updated? It seems to currently require restarting the game.

DaysSky commented:
stale
